frontend-v2/src/features/data/MapDosing.tsx

```diff
@@ -25,6 +25,12 @@ const MapDosing: FC<IMapDosing> = ({
   state,
   notificationsInfo,
 }: IMapDosing) => {
+  const amountField = state.fields.find(
+    (field) => state.normalisedFields.get(field) === "Amount",
+  );
+  // The amount field will be created if it doesn't exist, so this needs to be a 
+  // persistent state
+  const [hasDosingRows, _setHasDosingRows] = useState<boolean>(amountField !== undefined);
```

Comment:
A persistent reference might be easier to understand here:
```suggestion
  const amountFieldRef = useRef(state.fields.find(
    (field) => state.normalisedFields.get(field) === "Amount",
  ));
  const amountField = amountFieldRef.current;
  const hasDosingRows = amountField !== undefined;
```

Reply:
That should create `amountField` once, but not update it even if an amount field is later added to the CSV (which I assume is the case you're trying to catch here?)

Reply:
I've opened #697 to do the same thing, but with `useRef` instead of `useState`.

Reply:
great, thanks, I had not used `useRef` before. I'll close this and use #697 instead

Reply:
I'm still seeing a bug on that branch, if I upload a CSV without any dosing columns, so probably best to wait before merging it.

Reply:
I think I was missing a check for empty Amount column values. Something like:
```js
state.data.every((row) => row[amountField] !== "."
```